src/main/kotlin/app/cash/licensee/dependencyGraph.kt

```diff
@@ -163,82 +162,37 @@ private fun loadDependencyCoordinates(
 }
 
 internal fun loadPomInfo(
-  pomDocument: Document,
-  getParentPomDocument: (DependencyCoordinates) -> Document,
+  pom: Model,
+  getRawModel: (String) -> Model?,
 ): PomInfo {
-  var licensesNode: Node? = null
-  var scmNode: Node? = null
-  var parentNode: Node? = null
-  var artifactName: String? = null
-  for (childNode in pomDocument.documentElement.childNodes) {
-    when (childNode.nodeName) {
-      "licenses" -> licensesNode = childNode
-      "scm" -> scmNode = childNode
-      "parent" -> parentNode = childNode
-      "name" -> artifactName = childNode.textContent
-    }
+  val parentRawModel = pom.parent?.let {
+    getRawModel("${it.groupId}:${it.artifactId}:${it.version}")
   }
-
-  val licenses = mutableSetOf<PomLicense>()
-  if (licensesNode != null) {
-    for (licenseNode in licensesNode.childNodes) {
-      if (licenseNode.nodeName == "license") {
-        var name: String? = null
-        var url: String? = null
-        for (propertyNode in licenseNode.childNodes) {
-          when (propertyNode.nodeName) {
-            "name" -> name = propertyNode.textContent
-            "url" -> url = propertyNode.textContent
-          }
-        }
-        if (name != null || url != null) {
-          licenses += PomLicense(name, url)
-        }
-      }
+  val pomScm: Scm? = pom.scm
+  val url = if (parentRawModel != null) {
+    // https://maven.apache.org/ref/3.6.1/maven-model-builder/
+    // When depending on a parent, Maven adds a /artifactId to the url of the child,
+    // if the pom file does not opt out of this behavior:
+    // <scm child.scm.url.inherit.append.path="false">
+    // We don't want to handle the /artifactId, so we use the parent clean url instead.
+    val parentScm: Scm? = parentRawModel.scm
+    when (parentScm?.childScmUrlInheritAppendPath?.toBoolean()) {
```

Comment:
This is actually a workaround to keep the current behavior of licensee. According to the maven specs, the current extracted scm URL is wrong if you use a parent pom. Do you want to correct the behavior of licensee regarding the scm URL? If so, I will remove this workaround and change the tests.

Reply:
Damn that is tricky. On one hand, the SCM URL is useful for linking and we probably want the proper base URL. On the other hand, I don't want to be in the business of massaging a bunch of data.

Let's at least merge as-is and preserve existing behavior. I think long-term I'd like to figure out a way to educate users and figure out how to switch this off.